pysteps/nowcasts/steps.py

```diff
@@ -33,6 +33,873 @@
+            )
+
+        nowcast_utils.print_corrcoefs(self.gamma)
+
+        # Adjust the lag-2 correlation coefficient if AR(2) model is used
+        if self.ar_order == 2:
+            for i in range(self.n_cascade_levels):
+                self.gamma[i, 1] = autoregression.adjust_lag2_corrcoef2(
+                    self.gamma[i, 0], self.gamma[i, 1]
+                )
+
+        # Estimate the parameters of the AR model using autocorrelation coefficients
+        self.phi = np.empty((self.n_cascade_levels, self.ar_order + 1))
+        for i in range(self.n_cascade_levels):
+            self.phi[i, :] = autoregression.estimate_ar_params_yw(self.gamma[i, :])
+
+        nowcast_utils.print_ar_params(self.phi)
+
+        # Discard all except the last ar_order cascades for AR model
+        self.precip_cascades = [
+            self.precip_cascades[i][-self.ar_order :]
+            for i in range(self.n_cascade_levels)
+        ]
+
+        # Stack the cascades into a list containing all ensemble members
+        self.precip_cascades = [
+            [self.precip_cascades[j].copy() for j in range(self.n_cascade_levels)]
+            for _ in range(self.n_ens_members)
+        ]
+
+        # Initialize random generators if noise_method is provided
+        if self.noise_method is not None:
+            self.randgen_prec = []
+            self.randgen_motion = []
+
+            for _ in range(self.n_ens_members):
+                # Create random state for precipitation noise generator
+                rs = np.random.RandomState(self.seed)
+                self.randgen_prec.append(rs)
+                self.seed = rs.randint(0, high=int(1e9))  # Update seed after generating
+
+                # Create random state for motion perturbations generator
+                rs = np.random.RandomState(self.seed)
+                self.randgen_motion.append(rs)
+                self.seed = rs.randint(0, high=int(1e9))  # Update seed after generating
+        else:
+            self.randgen_prec = None
+            self.randgen_motion = None
+        print("AR model and noise applied to precipitation cascades.")
+
+    def _initialize_velocity_perturbations(self):
+        """
+        Initialize the velocity perturbators for each ensemble member if the velocity
+        perturbation method is specified.
+        """
+        if self.velocity_perturbation_method is not None:
+            init_vel_noise, generate_vel_noise = noise.get_method(
+                self.velocity_perturbation_method
+            )
+
+            self.velocity_perturbations = []
+            for j in range(self.n_ens_members):
+                kwargs = {
+                    "randstate": self.randgen_motion[j],
+                    "p_par": self.velocity_perturbation_kwargs.get(
+                        "p_par", self.velocity_perturbation_parallel
+                    ),
+                    "p_perp": self.velocity_perturbation_kwargs.get(
```

Comment:
```suggestion
                    "parameter_perpendicular": self.velocity_perturbation_kwargs.get(
```
Not sure if this suggestion is the best name, what do you think?